server/sonar-server/src/main/java/org/sonar/server/settings/ws/SetAction.java

```diff
@@ -126,6 +145,67 @@ public void handle(Request request, Response response) throws Exception {
     response.noContent();
   }
 
+  private void doHandle(DbSession dbSession, SetRequest request) {
+    Optional<ComponentDto> component = searchComponent(dbSession, request);
+    checkPermissions(component);
+
+    PropertyDefinition definition = propertyDefinitions.get(request.getKey());
+
+    if (!request.getFieldsValues().isEmpty()) {
+      doHandlePropertySet(dbSession, request, definition, component);
+    } else {
+      validate(request, component);
+      dbClient.propertiesDao().insertProperty(dbSession, toProperty(request, component));
+      if (!component.isPresent()) {
+        settingsChangeNotifier.onGlobalPropertyChange(request.getKey(), persistedValue(request));
+      }
+    }
+  }
+
+  private void doHandlePropertySet(DbSession dbSession, SetRequest request, @Nullable PropertyDefinition definition, Optional<ComponentDto> component) {
+    validatePropertySet(request, definition, component);
+
+    Set<String> fieldKeys = definition.fields().stream()
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) NullPointerException might be thrown as 'definition' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2259)